### PR TITLE
Refactor ZineScaffold to create project and issue

### LINKED ISSUE #130: Support new article and issue
`zine new --article` and `zine new --issue` will create a new article and issue respectively.

### PR DESCRIPTION
**Pull request by Folyd**
Partially resolve #130.

### CODE REVIEW

File: src/engine.rs [hunk 1]
```diff
@@ -191,7 +191,7 @@ impl ZineEngine {
 
         // Copy builtin static files into dest static dir.
         let dest_static_dir = self.dest.join("static");
-        fs::create_dir_all(&dest_static_dir)?;
+        fs::create_dir_all(dest_static_dir)?;
 
         #[cfg(not(debug_assertions))]
         include_dir::include_dir!("static").extract(dest_static_dir)?;
```
Comment (l1ch40) [inline src/engine.rs:197]: `dest_static_dir` value used here after move

File: src/new.rs [hunk 2]
```diff
@@ -1,62 +1,138 @@
...
+    // Use zine.toml to find root path
+    let (source, mut zine) = crate::locate_root_zine_folder(env::current_dir()?)?
+        .with_context(|| "Failed to find the root zine.toml file".to_string())?;
+    zine.parse_issue_from_dir(&source)?;
+
+    let author = run_command("git", &["config", "user.name"])?;
+    let next_issue_number = zine.issues.len() + 1;
+    let issue_dir = prompt_default(
+        "What is your issue directory name?",
+        format!("issue{next_issue_number}"),
```
Comment (l1ch40) [inline src/new.rs:121]: with `-` : `issue-{next_issue_number}`, like before it.

File: src/new.rs [hunk 3]
```diff
@@ -1,62 +1,138 @@
...
+    )?;
+    let issue_number = prompt_default("What is your issue number?", next_issue_number)?;
+    let issue_title = prompt_default(
+        "What is your issue title?",
+        format!("Issue-{next_issue_number}"),
```
Comment (l1ch40) [inline src/new.rs:126]: using single space instead of `-` : `Issue {next_issue_number}`, like before it.

**Comment by l1ch40** [comment]
Is this feature only `issue` but not `article`?

File: src/engine.rs (hunk 1 above)
Comment (Folyd) [reply src/engine.rs:197]: You are right, I'm confused why the CI didn't catch it 😕

**Comment by Folyd** [comment]
> Is this feature only `issue` but not `article`?

Yes, we can submit another PR to support `zine new --article`.

File: src/main.rs [hunk 4]
```diff
@@ -91,6 +96,9 @@ enum Commands {
     New {
         /// The project name.
         name: Option<String>,
+        /// New issue.
+        #[arg(short)]
+        issue: bool,
```
Comment (l1ch40) [inline src/main.rs:101]: can we add long command?

Comment (Folyd) [reply src/main.rs:101]: Sure, I just forgot it, thanks for reminding me.

**Pull request merged**